### PR TITLE
Make PandA hdf resource_path relative to root

### LINKED ISSUE #379: PandA resource_path includes root
https://github.com/bluesky/ophyd-async/blob/b59d1e21c91a2697c36e519393f69a29b7601978/src/ophyd_async/panda/writers/_panda_hdf_file.py#L33

PandA hdf resource_path is entire path: e.g.:

```json
    {
        "name": "stream_resource",
        "doc": {
            "uid": "8c37d78f-2f2f-47b5-ba74-b66831d0b962",
            "data_key": "panda1-pcap-ts_start-Value",
            "spec": "AD_HDF5_SWMR_SLICE",
            "root": "/dls/i22/data/2024/cm37271-2/bluesky",
            "resource_path": "/dls/i22/data/2024/cm37271-2/bluesky/i22-3-panda1.h5",
            "resource_kwargs": {
                "name": "panda1-pcap-ts_start-Value",
                "block": "pcap",
                "path": "PCAP-TS_START-Value",
                "multiplier": 1,
                "timestamps": "/entry/instrument/NDAttributes/NDArrayTimeStamp"
            },
            "path_semantics": "posix",
            "run_start": "0362ddf7-8f09-4539-9288-eb303e66c059"
        }
    }
```

which doesn't match with the expected `root / resource_path = absolute_path`

```json
    {
        "name": "stream_resource",
        "doc": {
            "uid": "f43ec9cc-8b7f-4579-897d-991bbb552364",
            "data_key": "i0",
            "spec": "AD_HDF5_SWMR_SLICE",
            "root": "/dls/i22/data/2024/cm37271-2/bluesky",
            "resource_path": "i22-3-i0-hdf.h5",
            "resource_kwargs": {
                "path": "/entry/data/data",
                "multiplier": 1,
                "timestamps": "/entry/instrument/NDAttributes/NDArrayTimeStamp"
            },
            "path_semantics": "posix",
            "run_start": "0362ddf7-8f09-4539-9288-eb303e66c059"
        }
    }
```

### PR DESCRIPTION
**Pull request by DiamondJoseph**
Fixes #379

### CODE REVIEW

**Comment by DiamondJoseph** [comment]
Tagging @mrakitin to make sure this won't impact NA-side PandA filewriting expectations?

File: src/ophyd_async/panda/writers/_hdf_writer.py [hunk 1]
```diff
@@ -203,9 +203,11 @@ async def collect_stream_docs(
         # TODO: fail if we get dropped frames
         if indices_written:
             if not self._file:
+                dir = Path(await self.panda_device.data.hdf_directory.get_value())
+                file = Path(await self.panda_device.data.hdf_file_name.get_value())
```
Comment (coretl) [inline src/ophyd_async/panda/writers/_hdf_writer.py:207]: There is already a PV that gives this:
https://github.com/PandABlocks/PandABlocks-ioc/blob/8e25d64cc97cfbf242a84c6ccbe600a39ed41c41/src/pandablocks_ioc/_hdf_ioc.py#L411-L413

Comment (coretl) [reply src/ophyd_async/panda/writers/_hdf_writer.py:207]: so
```python
    hdf_full_file_path: SignalR[str]
```
could be added alongside:
https://github.com/bluesky/ophyd-async/blob/b59d1e21c91a2697c36e519393f69a29b7601978/src/ophyd_async/panda/_common_blocks.py#L11-L12

File: src/ophyd_async/panda/writers/_panda_hdf_file.py [hunk 2]
```diff
@@ -30,7 +30,7 @@ def __init__(
                 spec="AD_HDF5_SWMR_SLICE",
                 root=str(directory_info.root),
                 data_key=ds.name,
-                resource_path=(f"{str(directory_info.root)}/{full_file_name}"),
+                resource_path=str(full_file_path.relative_to(directory_info.root)),
```
Comment (evalott100) [inline src/ophyd_async/panda/writers/_panda_hdf_file.py:33]: A test is erroring here because `root` is absolute, however `full_file_path` is "`.`".

**Review by evalott100 — CHANGES_REQUESTED** [review]

**Comment by DiamondJoseph** [comment]
Abandoning to prevent merge conflicts with #397 which should resolve this issue

**Pull request closed without merging**